test/vagrant-plugin/scenarios/molecule/box_url/molecule.yml

```diff
@@ -8,8 +8,8 @@ driver:
 platforms:
   - name: instance
     box: centos/stream9
-    box_url: "https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-Vagrant-9-20230704.1.x86_64.vagrant-libvirt.box"
-    box_download_checksum: "8a9fe7e8083421047e5404131daa007b4c1bcc466f3d9b29d0e7c7e432891a0e"
+    box_url: "https://cloud.centos.org/centos/8/vagrant/x86_64/images/CentOS-8-Vagrant-8.4.2105-20210603.0.x86_64.vagrant-libvirt.box"
```

Comment:
the `box: centos/stream9` but the `box_url` points to an old Centos 8.

but why not https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-Vagrant-9-20240930.0.x86_64.vagrant-virtualbox.box?

Reply:
oh, no vbox specific box please... well, one could argue that we're using libvirt box, breaking tests for vbox ...
I noticed this bug too and my idea was to do some ugly hack like using python httpserver  to serve the testbox but -ENOTIME.

Reply:
my mistake, didn't realize it was a `vagrant-virtualbox`, I just took one as an example.
https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-Vagrant-9-20240930.0.x86_64.vagrant-libvirt.box (https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-Vagrant-9-20240930.0.x86_64.vagrant-libvirt.box.SHA256SUM)

Reply:
The point is this box will "expire" in less than a year, while the Centos-8 will be there "forever".  There's 5 v8 libvirt boxes now, the oldest from 2019, but even if they decide to clean up at some point, there's a good chance they will keep the most recent one just for reference.

Reply:
> the `box: centos/stream9` but the `box_url` points to an old Centos 8.

Interestingly, I got a `centos/stream9` box in Vagrant storage, so it seems that when `config.vm.box_url` is specified, `config.vm.box` (normally used to identify boxes in Vagrant Cloud) will be retained to name the vm in local Vagrant repo.

I ran the test with  `box: centos/8-legacy` and sure enough I got a `centos/8-legacy` in local storage.

The Vagrant [documentation](https://developer.hashicorp.com/vagrant/docs/vagrantfile/machine_settings) is unclear on this matter:

> - config.vm.box (string) - This configures what box the machine will be brought up against. The value here should be the name of an installed box or a shorthand name of a box in HashiCorp's Vagrant Cloud.
> - config.vm.box_url (string, array of strings) - The URL that the configured box can be found at. If config.vm.box is a shorthand to a box in HashiCorp's Vagrant Cloud then this value does not need to be specified. Otherwise, it should point to the proper place where the box can be found if it is not installed. This can also be an array of multiple URLs. The URLs will be tried in order.

---

So I can update the `box_url` entry to something like `centos/8-legacy`, or update `box_url`, `box_checksum` to latest Centos 9.

As you wish.

Reply:
I vote to update `box_url` and `box_download_checksum` to latest Centos 9, so we use current up-to-date images.

Reply:
Updated to `CentOS-Stream-Vagrant-9-20240930.0.x86_64.vagrant-libvirt.box`. Test passes (but see discussion below).